geometry/grassmann.hpp

```diff
@@ -255,33 +257,23 @@ Multivector<Scalar, Frame, rank> operator-(
     Multivector<Scalar, Frame, rank> const& left,
     Multivector<Scalar, Frame, rank> const& right);
 
-template<typename Scalar, typename Frame, int rank>
-Multivector<Scalar, Frame, rank> operator*(
-    double left,
-    Multivector<Scalar, Frame, rank> const& right);
-template<typename Scalar, typename Frame, int rank>
-Multivector<Scalar, Frame, rank> operator*(
-    Multivector<Scalar, Frame, rank> const& left,
-    double right);
-template<typename Scalar, typename Frame, int rank>
-Multivector<Scalar, Frame, rank> operator/(
-    Multivector<Scalar, Frame, rank> const& left,
-    double right);
-
-template<typename LDimension, typename RScalar, typename Frame, int rank>
-Multivector<Product<Quantity<LDimension>, RScalar>, Frame, rank>
-operator*(Quantity<LDimension> const& left,
+template<typename LScalar, typename RScalar, typename Frame, int rank,
+         typename = std::enable_if_t<is_quantity<LScalar>::value>>
+Multivector<Product<LScalar, RScalar>, Frame, rank>
```

Comment:
Note that if you want this to work with wide quantities, a `Quantity operator*(Wide<Quantity>, Quantity)` has to be declared.

This is OK: we can declare the operators, which allows their usage in a `decltype`, and put a failing dependent `static_assert` in their definition to prevent them from being used, e.g., https://godbolt.org/g/DbUP24.

It makes things easier to understand and gives us good diagnostics, so this is probably the way to go.

Reply:
Ack.